### PR TITLE
Snapshots on ISPyB (fix #399)

### LINKED ISSUE #399: Snapshots on ISPyB
**User story**
As a user, I want to see snapshots in ISPyB

**Acceptance criteria**
Find my snapshots in ISPyB

### PR DESCRIPTION
**Pull request by mguijarr**
Fix #399 

It's a WIP because the code is untested. I don't know if it runs. But 90% of the thing is there...

I will remove WIP when I will have tested it here.

Meanwhile your comments are welcome.

For the record, there is some funny code to be able to work both with `AbstractMultiCollect` and `AbstractCollect`. When we will finally clean all the mess of abstract objects and their implementations, the code will be cleaned...

### CODE REVIEW

File: mxcube3/ui/components/SampleView/SampleControls.js [hunk 1]
```diff
@@ -12,14 +12,20 @@ export default class SampleControls extends React.Component {
...
+    this.doTakeSnapshot = this.doTakeSnapshot.bind(this);
     this.setZoom = this.setZoom.bind(this);
     this.setApertureSize = this.setApertureSize.bind(this);
     this.toggleFrontLight = this.toggleLight.bind(this, 'FrontLight');
     this.toggleBackLight = this.toggleLight.bind(this, 'BackLight');
     this.toggleCentring = this.toggleCentring.bind(this);
   }
 
+  componentDidMount() {
+    window.takeSnapshot = this.doTakeSnapshot;
```
Comment (meguiraun) [inline mxcube3/ui/components/SampleView/SampleControls.js]: why linking to the `window` element? (not that I know an alternative ...)

**Comment by meguiraun** [comment]
as far as I can tell looks fine, but I have just gone through the code without testing.

**Comment by mguijarr** [comment]
Thanks - I will test tomorrow when beamline is free, then remove the WIP flag.

**Comment by mguijarr** [comment]
Tested on ID30A3. Ready to merge.

**Comment by amilan** [comment]
Hi Matias, 

At this moment we are not able to test it, but Mikel has reviewed the code and it looks OK, so we can merge it.

Later on, when @bolmsten will be back, he can review it again if he wants.

Thanks!

**Pull request merged**